src/azul/service/responseobjects/elastic_request_builder.py

```diff
@@ -154,8 +142,33 @@ def _create_aggregate(self, filters, facet_config, agg):
         # value back in
         if excluded_filter is not None:
             filters[facet_config[agg]] = excluded_filter
+        # Store a copy of the ES field name aggregated in each Terms aggregate's 'meta' field
+        self.add_meta_field_to_aggregate(aggregate)
         return aggregate
 
+    @classmethod
+    def add_meta_field_to_aggregate(cls, aggregate):
+        """
+        Update the aggregate to add a meta field containing the full name of the ES field being aggregated
+        """
+        # If we have a Filter object iterate over the contents to find any Terms objects
+        if isinstance(aggregate, Filter):
+            for agg_name in aggregate:
+                agg = aggregate[agg_name]
+                if isinstance(agg, Terms):
+                    cls.add_meta_field_to_aggregate(agg)
+        # If we have a Terms object copy out the 'field' value and stuff it in the 'meta' field
+        elif isinstance(aggregate, Terms):
+            field = aggregate.field
+            suffix = '.keyword'
```

Comment:
Try splitting here and stashing the tuple like we talked about. I'd like to be convinced that that is not easier.